src/execution/operator/schema/physical_create_type.cpp

```diff
@@ -30,6 +56,69 @@ void PhysicalCreateType::GetData(ExecutionContext &context, DataChunk &chunk, Gl
 	if (state.finished) {
 		return;
 	}
+
+	if (IsSink()) {
+		D_ASSERT(info->type == LogicalType::INVALID);
+
+		auto &g_sink_state = (CreateTypeGlobalState &)*sink_state;
+		auto &collection = g_sink_state.collection;
+		ColumnDataScanState scan_state;
+		collection.InitializeScan(scan_state);
+
+		DataChunk scan_chunk;
+		collection.InitializeScanChunk(scan_chunk);
+
+		idx_t total_row_count = collection.Count();
+		Vector result(LogicalType::VARCHAR, total_row_count);
+		auto result_ptr = FlatVector::GetData<string_t>(result);
+		auto &res_validity = FlatVector::Validity(result);
+
+		idx_t offset = 0;
+		while (collection.Scan(scan_state, scan_chunk)) {
+			idx_t src_row_count = scan_chunk.size();
+			auto &src_vec = scan_chunk.data[0];
+			D_ASSERT(src_vec.GetVectorType() == VectorType::FLAT_VECTOR);
+			D_ASSERT(src_vec.GetType().id() == LogicalType::VARCHAR);
+
+			auto src_ptr = FlatVector::GetData<string_t>(src_vec);
+			auto &src_validity = FlatVector::Validity(src_vec);
+
+			if (!src_validity.AllValid()) {
+				// lazy initialize
+				if (res_validity.AllValid()) {
```

Comment:
Enums should not have NULL values. Instead an error should be thrown when a NULL value is encountered.

This is also overly optimized - in general I would expect this code not to be run with more than a few thousand rows.